airflow/auth/managers/simple/simple_auth_manager.py

```diff
@@ -78,20 +79,23 @@ class SimpleAuthManager(BaseAuthManager):
     :param appbuilder: the flask app builder
     """
 
-    # File that contains the generated passwords
-    GENERATED_PASSWORDS_FILE = (
-        AIRFLOW_ROOT_PATH / "generated" / "simple_auth_manager_passwords.json.generated"
-    )
-
     # Cache containing the password associated to a username
     passwords: dict[str, str] = {}
 
+    @staticmethod
+    def get_generated_password_file() -> Path:
+        filename = "simple_auth_manager_passwords.json.generated"
+        if os.getenv("AIRFLOW_AUTH_MANAGER_CREDENTIAL_DIRECTORY"):
+            return Path(f"{os.getenv('AIRFLOW_AUTH_MANAGER_CREDENTIAL_DIRECTORY')}/{filename}")
+        else:
+            return AIRFLOW_ROOT_PATH / "files" / filename
```

Comment:
I think we cannot use this AIRFLOW_ROOT_PATH. The `hatch_build.py` where it is imported from is not available when airflow is installed as package - it's only available from sources. So we should have a bit "saner" default. I thi

Reply:
I think we should handle (similarly as with Providers Manager) - two cases, one where airflow (and providers in case of provider's manager) are installed from sources, and another when Airflow is installed as package. In the second case I think good default is somewhere in AIRFLOW_HOME or "HOME/airflow" - this is for example where we create sqlite database when we use sqlite as backend.

Reply:
I can see that the standalone command uses `password_path = os.path.join(AIRFLOW_HOME, "standalone_admin_password.txt")` to store the password. Should not we do the same? Like: password_path = os.path.join(AIRFLOW_HOME, "simple_auth_manager_passwords.json.generated")?

Reply:
I meant with no branches, use the same path regardless of the environment. I am just trying to make it simple, I feel like it is getting overly complex for something that should not be

Reply:
But now I remember this [comment](https://github.com/apache/airflow/pull/42004#discussion_r1750302628). Should we still consider `AIRFLOW_HOME` then?

Reply:
I am trying to wrap my head around all this logic .. Is there any code in place I can take inspiration from? I dont quite know how to figure if Airflow is installed from sources or as a package

Reply:
I changed the logic, it is quite simple now:
- if `AIRFLOW_AUTH_MANAGER_CREDENTIAL_DIRECTORY` is defined, the passwords are saved there. This is the case when run with Breeze (I set it in `entrypoint_ci.sh`)
- If not defined, save the file in `AIRFLOW_HOME`, unless I misunderstood something, I dont think we should make a difference whether Airflow is installed with sources or as a package. In both cases, saving the passwords in `AIRFLOW_HOME` is fine

WDYT?